src/server/worker/api_server.go

```diff
@@ -1716,9 +1711,10 @@ func (a *APIServer) cancelCtxIfJobFails(jobCtx context.Context, jobCancel func()
 					return fmt.Errorf("job state watch error: %v", e.Err)
 				}
 			case <-jobCtx.Done():
-				break
+				break Outer
```

Comment:
This was actually a big problem that it caught, the `break` was being applied to the select (ie doing nothing), rather than the for loop, so we were never actually breaking out of the loop.